### ISSUE TITLE
UndocumentedPublicClass in conjunction with typealias

### ISSUE THREAD

**Issue by vanniktech** [arturbosch/detekt#193]
I've got a few cases where I've got `const` or `typealias` before my class declaration which does get flagged by `UndocumentedPublicClass`:

```
/**
 * @author niklasbaudy / vanniktech
 * @since 04.07.17
 */
typealias Foo = Int

interface Bar {
}
```

Should this get flagged? In the case of `typealias` it needs to be outside of the `interface` declaration.

**Comment by arturbosch** [comment]
Yep, because the comment is clearly over typealias in the AST. You need to move the typealias belowe the interface ^^

**Comment by vanniktech** [comment]
Yup I can understand this although I'd look weird if it would be this way:

```kotlin
typealias Foo = Int

/**
 * @author niklasbaudy / vanniktech
 * @since 04.07.17
 */
interface Bar {
}
```

**Comment by arturbosch** [comment]
There are also imports and package so not that odd ;) closing this?

**Comment by vanniktech** [comment]
Sure :D
